### PR TITLE
Brand new "Learn" section

### LINKED ISSUE #313: Add a CookBook section
I've been thinking about this for a while now and adding a CookBook/Micro Tutorial section makes sense. The Guides are already very elaborate, but it would help users to show them how to use Lucky with features that are too specific, opinionated or beyond the scope of the Guides, although still Lucky-specific.

I think we all have great solutions lying around that would help users jump-start their projects. For example, I created a [Mailjet](https://github.com/tilishop/mailjet.cr)-powered newsletter subscription form with a few spam-bot traps. Or a flexible, well-structured [I18n.cr](https://github.com/vladfaust/i18n.cr) implementation that uses the current class name + namespace for its nested keys, so you don't have to think about structuring your locale files. But there are also shards like @confact's [`lucky_can`](https://github.com/confact/lucky_can) or my [`lucky_svg_sprite`](https://github.com/tilishop/lucky_svg_sprite.cr). 

Those things don't belong in the Guides, but would still be very valuable to have. They could be posted on DEV.to or Medium, but they would disappear in the noise quickly. Better even, they could have a place on those platforms and in the site somewhere as well.

It's true that it adds maintenance, but that's easily solved. We can have an app containing all those implementations which can be run through Travis or Github Actions at every release, just to see if everything is still functioning correctly.

Food for thought. :blush:

### PR DESCRIPTION
**Pull request by jwoertink**
This PR was built by pair-programming with @stephendolan. It was pretty awesome!

Fixes #313 
Related #204 (Will eventually link to these from the Ecosystem page)
Related #304 
Related #280 (This can go in the community section)
Related #414 (This can also go in the community section or links in the awesome section)

Screen Shot 2021-04-17 at 3 09 05 PM

### CODE REVIEW

File: src/pages/learn/awesome_lucky/index_page.cr [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class Learn::AwesomeLucky::IndexPage < PageLayout
+  def hero_content
+    render_hero_content
+  end
+
+  def content
+    # https://github.com/andrewmcodes/awesome-lucky
+    div class: "py-4 container text-center" do
+      text "Coming soon..."
```
Comment (jwoertink) [inline src/pages/learn/awesome_lucky/index_page.cr]: These pages display a "Coming soon..." for now which will allow us to do these in separate PRs

File: src/pages/learn/community/index_page.cr [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class Learn::Community::IndexPage < PageLayout
+  def hero_content
+    render_hero_content
+  end
+
+  def content
+    # Discord chat, twitter, maybe mention the t-shirt giveaway...
```
Comment (jwoertink) [inline src/pages/learn/community/index_page.cr:7]: Just some ideas, but nothing set in stone.

File: src/pages/learn/index_page.cr [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    render_hero_content
+  end
+
+  def content
+    div class: "md:-mt-12 md:rounded-lg bg-gray-200 overflow-hidden shadow divide-y divide-gray-200 sm:divide-y-0 sm:grid sm:grid-cols-2 sm:gap-px" do
+      div class: "md:rounded-tl-lg md:rounded-tr-lg sm:rounded-tr-none relative group bg-white p-6 focus-within:ring-2 focus-within:ring-inset focus-within:ring-indigo-500" do
+        div do
+          span class: "rounded-lg inline-flex p-3 bg-green-50 text-green-700 ring-4 ring-white" do
+            tag "svg", class: "h-6 w-6", fill: "none", stroke: "currentColor", viewBox: "0 0 24 24", xmlns: "http://www.w3.org/2000/svg" do
+              tag "path", d: "M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z", stroke_linecap: "round", stroke_linejoin: "round", stroke_width: "2"
```
Comment (jwoertink) [inline src/pages/learn/index_page.cr]: SVG icons provided by https://heroicons.com/

File: src/components/shared/header.cr [hunk 4]
```diff
@@ -53,7 +53,7 @@ class Shared::Header < BaseComponent
   end
 
   private def nav_links
-    nav_link("Guides", Guides::GettingStarted::Installing.path)
+    nav_link("Learn", Learn::Index.path)
```
Comment (jwoertink) [inline src/components/shared/header.cr]: It may be redundant, but we could also have both links... 🤔

File: src/pages/page_layout.cr [hunk 5]
```diff
@@ -18,6 +18,11 @@ abstract class PageLayout
 
       body class: "font-sans text-grey-darkest leading-tight bg-grey-lightest" do
         mount Shared::Header, @context.request
+        if responds_to?(:hero_content)
+          div class: "hero_content" do
+            render_if_defined :hero_content
+          end
+        end
```
Comment (jwoertink) [inline src/pages/page_layout.cr:25]: The default page puts all of the content inside of a `container`. This section allows us to do that gradient that spans 100%

File: src/actions/learn/index.cr [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class Learn::Index < BrowserAction
+  get "/learn" do
+    html Learn::IndexPage, title: "Learn the Lucky Framework"
```
Comment (edwardloveall) [inline src/actions/learn/index.cr:3]: How do you feel about the title of all these pages going in the code for the page?

File: src/components/shared/header.cr (hunk 4 above)
Comment (edwardloveall) [reply src/components/shared/header.cr]: I do like having a direct links to docs. Learn is great for "this is my first time" vs docs being good for anyone of experience getting to reference quickly. Does it look bad with both? If not, I'd suggest that. If so, we could a drop down or put "Learn" prominently on the homepage instead? The idea being that the nav is for people looking for something particular, but the homepage is more general information if you're trying to understand what lucky is.

File: src/pages/learn/index_page.cr (hunk 3 above)
Comment (edwardloveall) [reply src/pages/learn/index_page.cr]: Would it be worth it to abstract these icons to methods so this page is easier to read? The ⬈ in particular since it's repeated so many times.

File: src/pages/learn/index_page.cr [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+        end
+        div class: "mt-8" do
+          h3 class: "text-lg font-medium" do
+            a class: "focus:outline-none", href: "https://github.com/andrewmcodes/awesome-lucky#readme", target: "_blank" do
+              span aria_hidden: "true", class: "absolute inset-0"
+              text "Awesome Lucky"
+            end
+          end
+          para class: "mt-2 text-sm text-gray-500" do
+            text "Awesome Lists are lists of very helpful resources within a particular community. This is our list for Awesome Lucky originally provided by the Awesome @andrewmcodes."
```
Comment (edwardloveall) [inline src/pages/learn/index_page.cr]:
```suggestion
            text "Awesome Lists are lists of helpful resources within a particular community. This is our list for Awesome Lucky originally provided by the Awesome @andrewmcodes."
```

**Review by edwardloveall — APPROVED** [review]
Thanks for putting this together. I had a few small suggestions.

File: src/actions/learn/index.cr (hunk 6 above)
Comment (jwoertink) [reply src/actions/learn/index.cr:3]: I think the idea Paul was going for here is that this makes the title required to be passed in to ensure you don't make a new page and forget to set the title. I think we can do it differently though.

File: src/components/shared/header.cr (hunk 4 above)
Comment (stephendolan) [reply src/components/shared/header.cr]: Yeah, I really like having both in the top navbar as well.

File: src/actions/learn/index.cr (hunk 6 above)
Comment (edwardloveall) [reply src/actions/learn/index.cr:3]: Yeah that's fair. I don't feel super strongly about it to be clear. I just found it odd that it's the action's responsibility to set the title for all pages. Felt like the page should own that. NBD

File: src/components/shared/header.cr (hunk 4 above)
Comment (stephendolan) [reply src/components/shared/header.cr]: Added the Guides link back for now!

**Review by stephendolan — APPROVED** [review]

**Comment by jwoertink** [comment]
Sweet! Thanks for finishing that up @stephendolan
